pina/solvers/pinn.py

```diff
@@ -108,25 +117,29 @@ def training_step(self, batch, batch_idx):
         :rtype: LabelTensor
         """
 
+        dataloader = self.trainer.train_dataloader
         condition_losses = []
-        condition_names = []
 
-        for condition_name, samples in batch.items():
+        condition_idx = batch['condition']
 
-            if condition_name not in self.problem.conditions:
-                raise RuntimeError('Something wrong happened.')
+        for condition_id in range(condition_idx.min(), condition_idx.max()+1):
 
-            condition_names.append(condition_name)
+            condition_name = dataloader.condition_names[condition_id]
             condition = self.problem.conditions[condition_name]
+            pts = batch['pts']
+
+            if len(batch) == 2:
+                samples = pts[condition_idx == condition_id]
+                loss = self._loss_phys(pts, condition.equation)
+            elif len(batch) == 3:
+                samples = pts[condition_idx == condition_id]
+                ground_truth = batch['output'][condition_idx == condition_id]
+                loss = self._loss_data(samples, ground_truth)
+            else:
+                raise ValueError("Batch size not supported")
```

Comment:
I am not sure what is going on here. Can you give me more details? Do you think we can make it easier so someone who is building a solver doesn't have to play with len(batch). Maybe a function in the utility, idk...